### PR TITLE
Coefficients for Cohen-Hubbard-Oesterwinter

### PR DESCRIPTION
**Pull request by pleroy**
and a fix for the order of the backward difference method.

### CODE REVIEW

File: integrators/cohen_hubbard_oesterwinter.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace principia {
+namespace integrators {
+namespace internal_cohen_hubbard_oesterwinter {
+
+using numerics::FixedVector;
+
+// Definition of a Cohen-Hubbard-Oesterwinter formula.  |order_| is the order of
+// the approximation, that is, the error on the derivative is h^(order + 1).
+// The formula requires |order_| values of the acceleration.  numerators[0]
+// corresponds to f"(x₀), numerators[1] to f"(x₋₁), etc.
```
Comment (eggrobin) [inline integrators/cohen_hubbard_oesterwinter.hpp]: Use ″.

File: integrators/cohen_hubbard_oesterwinter.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include "numerics/fixed_arrays.hpp"
+
+namespace principia {
+namespace integrators {
+namespace internal_cohen_hubbard_oesterwinter {
+
+using numerics::FixedVector;
+
+// Definition of a Cohen-Hubbard-Oesterwinter formula.  |order_| is the order of
+// the approximation, that is, the error on the derivative is h^(order + 1).
```
Comment (eggrobin) [inline integrators/cohen_hubbard_oesterwinter.hpp]: O()?

File: integrators/cohen_hubbard_oesterwinter.hpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// The formula requires |order_| values of the acceleration.  numerators[0]
+// corresponds to f"(x₀), numerators[1] to f"(x₋₁), etc.
+// TODO(phl): This struct shows up in many places.  Factor it out.
```
Comment (eggrobin) [inline integrators/cohen_hubbard_oesterwinter.hpp]: I don’t understand this TODO.

File: integrators/cohen_hubbard_oesterwinter.hpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+#include "numerics/fixed_arrays.hpp"
+
+namespace principia {
+namespace integrators {
+namespace internal_cohen_hubbard_oesterwinter {
+
+using numerics::FixedVector;
+
+// Definition of a Cohen-Hubbard-Oesterwinter formula.  |order_| is the order of
```
Comment (eggrobin) [inline integrators/cohen_hubbard_oesterwinter.hpp]: Cite the paper, document any notational differences from the paper (I think we're off-by-one compared to it?)

File: mathematica/cohen_hubbard_oesterwinter.wl [hunk 5]
```diff
@@ -39,17 +36,17 @@ With[
   {j, 0, n - 2}]]];
 
 
-{{LCM@@(Denominator/@#),#*LCM@@(Denominator/@#)}}&[\[Beta]]//TableForm
+{{LCM@@(Denominator/@#),#*LCM@@(Denominator/@#)}}&[\[Beta][14]]//TableForm
 
 
-(fs[-h,\[ScriptCapitalN]+3]-fs[-2h,\[ScriptCapitalN]+3])/h+h Sum[f2s[-i h,\[ScriptCapitalN]+3]\[Beta][[i]],{i,1,Length[\[Beta]]}]//Expand
+ReleaseHold[Hold[(fs[-h,\[ScriptCapitalN]+3]-fs[-2h,\[ScriptCapitalN]+3])/h+h Sum[f2s[-i h,\[ScriptCapitalN]+3]\[Beta][\[ScriptCapitalN]][[i]],{i,1,Length[\[Beta][\[ScriptCapitalN]]]}]]/.\[ScriptCapitalN]->14]//Expand
```
Comment (eggrobin) [inline mathematica/cohen_hubbard_oesterwinter.wl]: What is going on with this ReleaseHold and Hold?

File: integrators/cohen_hubbard_oesterwinter.hpp [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+// the approximation, that is, the error on the derivative is h^(order + 1).
+// The formula requires |order_| values of the acceleration.  numerators[0]
```
Comment (eggrobin) [inline integrators/cohen_hubbard_oesterwinter.hpp]: Document what the formula is, and what it approximates.

File: integrators/cohen_hubbard_oesterwinter.hpp (hunk 1 above)
Comment (pleroy) [reply integrators/cohen_hubbard_oesterwinter.hpp]: Done.

File: integrators/cohen_hubbard_oesterwinter.hpp (hunk 2 above)
Comment (pleroy) [reply integrators/cohen_hubbard_oesterwinter.hpp]: Done.

File: integrators/cohen_hubbard_oesterwinter.hpp (hunk 4 above)
Comment (pleroy) [reply integrators/cohen_hubbard_oesterwinter.hpp]: Done.

File: integrators/cohen_hubbard_oesterwinter.hpp (hunk 6 above)
Comment (pleroy) [reply integrators/cohen_hubbard_oesterwinter.hpp]: Done.

File: mathematica/cohen_hubbard_oesterwinter.wl (hunk 5 above)
Comment (pleroy) [reply mathematica/cohen_hubbard_oesterwinter.wl]: _Mathematica_ magic: (1) Hold prevents the functions like `fs` from being evaluated at a place (2) `N` is being substituted in the argument of `fs` et al. (3) releasing the hold causes the whole thing to evaluate.

File: integrators/cohen_hubbard_oesterwinter.hpp (hunk 3 above)
Comment (pleroy) [reply integrators/cohen_hubbard_oesterwinter.hpp]: We have 3 or 4 occurrences of a struct containing a `FixedVector` for numerators and a single value for the denominator.

**Pull request merged**